### ISSUE TITLE
urlCreator callback missing controller attribute

### ISSUE THREAD

**Issue by chaimleich** [yiisoft/yii2#11658]
yii\grid\ActionColumn allows the ability to custom-generate the action url using the urlCreator callback. The signature is as follows:
`'urlCreator' => function($action, $model, $key, $index) { ... }`

Compare this to the code for the createUrl method. Note that the createUrl method takes into account the 'controller' attribute, but this controller attribute is not passed to the urlCreator callback. As a result, the following code fails:
```
'urlCreator' => function($action, $model, $key, $index) {
    $params = is_array($key) ? $key : ['id' => (string) $key];
	// Option 1 - 500 error - $controller not an attribute of yii\web\View
    $params[0] = $this->controller ? $this->controller . '/' . $action : $action;
	// Option 2 - technically works but obviously has no effect
    $params[0] = $controller ? $controller . '/' . $action : $action;
    return Url::toRoute($params);
}
```

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.8
| PHP version      | 7.0
| Operating system | CentOS

**Comment by samdark** [comment]
I think it could be changed to pass more parameters.

**Comment by samdark** [comment]
Do you want to work on a pull request?

**Comment by chaimleich** [comment]
I won't have time to prepare a proper pull request for this in the coming weeks. I personally don't need this functionality; I just wanted to bring this the the attention of the Yii team.

**Comment by SamMousa** [comment]
@samdark Would the controller be added as the last argument to limit breaking BC?

If yes, feel free to assign to me.

**Comment by samdark** [comment]
Yes.

**Comment by klimov-paul** [comment]
I can see no benefit of adding `ActionColumn::controller` to the `urlCreator` callback.

`ActionColumn::controller` has no predefined or auto-determined value - it is always configured by the developer manually on use.
What is the sense to pass manually configured parameter to the manually specified callback?

**Comment by klimov-paul** [comment]
In particular:

```php
[
    'class' => ActionColumn::className(),
    'controller' => 'custom-controller',
    'urlCreator' => function ($action, $model, $key, $index) {
         // are you already forget what value is set 2 lines above?
    },
],
```

**Comment by klimov-paul** [comment]
Resolved by commit e0882a864d70af39852299833f0a359798572708
